src/pull-progress-indicator.js

```diff
@@ -0,0 +1,103 @@
+const logUpdate = require('log-update');
+const { green, cyan, yellow } = require('colorette');
+const emoji = require('node-emoji');
+
+/**
+ * Some status names are long; this map has shorter names!
+ */
+const SHORT_NAME_STATUS = {
+    'pulling fs layer': 'Preparing',
+    'download complete': 'Downloaded',
+    'pull complete': 'Completed',
+    'retry': 'Retrying'
+};
+
+/**
+ * Maximum column length for status 
+ */
+const STATUS_STRING_MAX_LENGTH = 25;
+
+/**
+ * Cached emoji, so it needn't be fetched repeatedly!
+ */
+const PACKAGE_EMOJI = emoji.get('package');
+
+
+/**
+ * Starts up a progress bar for the `docker pull` operation, parses the event stream from dockerode to get the progress info.
+ */
+class PullProgressIndicator {
+    constructor() {
+        this._chunkInfo = null;
+    }
+
+    /**
+     * Gets shortened status name, because some status names can be a wee bit long.
+     * 
+     * @param {string} status 
+     */
+    _getShortenedStatus(status) {
+        if (!status) return '';
+
+        // Special case for retries
+        if (status.includes('retry')) return SHORT_NAME_STATUS.retry;
+
+        return SHORT_NAME_STATUS[status.toLowerCase()] || status;
+    }
+
+    /**
+     * Uses `log-update` to update the progress information on the terminal
+     */
+    _redrawProgress() {
+        if (this._chunkInfo === null) {
+            // TODO: Should we persist this? https://www.npmjs.com/package/log-update#logupdatedone
+            return logUpdate.clear();
+        }
+
+        const chunkProgress = Object.keys(this._chunkInfo).map(id => {
+            const { status, progress } = this._chunkInfo[id];
+            const chunkPart = `${PACKAGE_EMOJI} ${yellow(id)}`;
+
+            // Padding because even columns are <3 
+            const statusPart = green(this._getShortenedStatus(status).padEnd(STATUS_STRING_MAX_LENGTH));
+
+            const progressBarPart = cyan(progress || '');
```

Comment:
@BrunnerLivio I know that you mentioned [`progress-estimator`](https://github.com/bvaughn/progress-estimator) in #3, but I noticed that `dockerode` returns an actual progress bar :scream_cat: 

The object looks like this: 

```
{
    "status": "Downloading",
    "progressDetail": {
      "current": 87345896,
      "total": 214829889
    },
    "progress": "[====================>                              ]  87.35MB/214.8MB",
    "id": "87fc2710b63f"
  }
```

I thought it would be easier to reuse that, than making my own progress bar via `progress-estimator`. What are your thoughts about this?

Reply:
Makes sense. Progress-estimator was just a suggestion, if docker does not return a accurate progress information, but this is much better :)